File: test-suite/java/suppression/suppression_buggy.java
// GH #91 suppression fixture (bead A7) for the Java module, twin of
// suppression_buggy_nomarkers.java (identical buggy code, no markers).
// Every native java finding below (Statement/PreparedStatement/ResultSet
// acquired outside try-with-resources, Runtime.exec — the path:line code
// samples printed by the module) carries an ubs:ignore marker in one of the
// documented arrangements, so scanning this file must report zero findings,
// while the nomarkers twin reproduces them.

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;

public class SuppressionBuggy {

    // Arrangement 1: previous-line marker.
    public void queryPrevLine(Connection conn) throws SQLException {
        // ubs:ignore -- fixture: marker on the line immediately above the finding
        Statement stmt = conn.createStatement();
        stmt.execute("SELECT 1");
    }

    // Arrangement 2: trailing marker on the flagged line itself.
    public void queryTrailing(Connection conn) throws SQLException {
        PreparedStatement ps = conn.prepareStatement("SELECT 1"); // ubs:ignore -- fixture: trailing marker
        ps.execute();
    }

    // Arrangement 3: multi-line statement, marker on a continuation line.
    public void queryMultiline(Statement stmt) throws SQLException {
        ResultSet rs =
            stmt.executeQuery("SELECT 1"); // ubs:ignore -- fixture: marker on a physical line of a multi-line statement
    }

    // Arrangement 4: formatter-relocated marker on the first line inside a block.
    public void queryInBlock(Connection conn) throws SQLException {
        if (conn != null) { Statement guarded = conn.createStatement();
            // ubs:ignore -- fixture: formatter moved the marker inside the block
        }
    }

    // Arrangement 5: rule-scoped markers (ubs:ignore[rule]). Java code-sample
    // lines carry no rule id, so these suppress through the runner's
    // same-line/previous-line path.
    public void ruleScopedPrevLine(Connection conn) throws SQLException {
        // ubs:ignore[java.resource.statement-no-close] -- fixture: rule-scoped marker above the finding
        Statement scoped = conn.createStatement();
        scoped.execute("SELECT 1");
    }

    public void ruleScopedTrailing(Connection conn) throws SQLException {
        PreparedStatement scopedPs = conn.prepareStatement("SELECT 1"); // ubs:ignore[java.resource.statement-no-close] -- fixture: rule-scoped trailing marker
        scopedPs.execute();
    }

    // Arrangement 6: Runtime.exec critical, previous-line marker.
    public void runShell(String userInput) throws java.io.IOException {
        // ubs:ignore -- fixture: marker on the line immediately above the finding
        Runtime.getRuntime().exec(userInput);
    }
}

File: test-suite/csharp/suppression/suppression_buggy.cs
// GH #91 suppression fixture (bead A7) for the C# module, twin of
// suppression_buggy_nomarkers.cs (identical buggy code, no markers).
// Every native csharp finding below (the path:line code-sample lines the
// module rg-based categories print natively) carries a suppression marker
// in one of the documented arrangements, so a scan of this file must report
// zero finding lines, while the nomarkers twin reproduces them.
// Note: comments here deliberately avoid apostrophes; the shared lexer masks
// single-quote strings for C#, so a stray apostrophe would swallow markers.
// Prose never spells out the marker token, which would parse as a marker.

using System;
using System.Diagnostics;
using System.Net.Http;
using System.Security.Cryptography;
using System.Threading;

public static class SuppressionBuggy
{
    // Arrangement 1: previous-line marker.
    public static void LaunchShellPrevLine(string userInput)
    {
        // ubs:ignore -- fixture: marker on the line immediately above the finding
        Process.Start("cmd.exe", "/C " + userInput);
    }

    // Arrangement 2: trailing marker on the flagged line itself.
    public static void LaunchToolTrailing()
    {
        var tool = new ProcessStartInfo("notepad.exe", "notes.txt");
        Process.Start(tool); // ubs:ignore -- fixture: trailing marker on the flagged line
    }

    // Arrangement 3: multi-line statement, marker on a continuation line.
    public static void LaunchShellMultiline(string userInput)
    {
        Process.Start("cmd.exe", "/C " +
            userInput); // ubs:ignore -- fixture: marker on a physical line of a multi-line statement
    }

    // Arrangement 4: formatter-relocated marker on the first line inside a block.
    public static void WeakHashInBlock()
    {
        using (var digest = MD5.Create())
        {   // ubs:ignore -- fixture: formatter moved the marker off the flagged line into the block
            _ = digest;
        }
    }

    // Arrangement 5: rule-scoped markers (rule id in square brackets). The
    // csharp module code-sample lines carry no rule id, so a previous-line
    // scope suppresses through the runner flat same-line/previous-line path,
    // while the trailing scope is honored by the interval engine itself (the
    // bracketed rule id shows up in the printed sample line).
    public static void RuleScopedPrevLine()
    {
        // ubs:ignore[cs.thread-sleep-blocks] -- fixture: rule-scoped marker above the finding
        Thread.Sleep(5);
    }

    public static void RuleScopedTrailing()
    {
        var client = new HttpClient(); // ubs:ignore[cs.httpclient-per-call] -- fixture: rule-scoped trailing marker
        _ = client;
    }
}
